### ISSUE TITLE
Would you like to work on this together?

### ISSUE THREAD

**Issue by Imtiaz08** [KyZEN1711/GNSS-Spoofing#1]

**Comment by Keshav1711** [comment]
I am working on this as a part of an RnD project at my Univ. I don't know much about GNSS so any help is appreciated.

**Comment by Imtiaz08** [comment]
I have done my Masters in GNSS and have worked on GNSS spoofing as my thesis. 

if you are trying to build a GNSS spoofer we can do this together. 

There's a MATLAB code that is about 70% complete. We can complete it by calculating the code and carrier phases and then generate the signal

**Comment by Keshav1711** [comment]
Wow, that is amazing. My end goal for this is to create a GNSS spoofer, I'd like to contribute to it.

**Comment by Keshav1711** [comment]
Can you send me the repo to you project?

**Comment by Imtiaz08** [comment]
Yes I can

**Comment by Imtiaz08** [comment]
You can find it here https://github.com/Imtiaz08/gps-sim

**Comment by Keshav1711** [comment]
Looks Great, can you explain to me how the code works, what all is left and how can I contribute towards it?

**Comment by Imtiaz08** [comment]
Mind if we have a zoom meeting or whatsapp chat? 

I can explain the whole scenario in a very basic manner but we can discuss this whole project in detail 

First of all the code extracts ephemeris data from the RINEX files, It then calculates the satellite position, then pseudorange and then the pseudo range is converted into code phases and carrier phases. 

After that, a GPS L1 C/A signal is constructed based on the data in each channel. 

The subframe data is generated according to the GPS ICD

**Comment by Keshav1711** [comment]
Yeah, we can have a google meet. https://meet.google.com/diq-rxtv-poj

**Comment by Keshav1711** [comment]
Let's setup a meet, how about 3pm IST tomorrow (26)
